shell/platform/android/io/flutter/embedding/android/AndroidTouchProcessor.java

```diff
@@ -172,11 +179,20 @@ public boolean onTouchEvent(@NonNull MotionEvent event, @NonNull Matrix transfor
       // It's important that we're sending the UP event last. This allows PlatformView
       // to correctly batch everything back into the original Android event if needed.
       addPointerForIndex(event, event.getActionIndex(), pointerChange, 0, transformMatrix, packet);
+
+      if (shouldRemovePointer) {
+        // Flutter's pointer handling synthesizes an add event for each new pointer, but does not
+        // synthesize remove events. Remove each pointer on it's corresponding ACTION_UP or
+        // ACTION_POINTER_UP if and only if the input device is touch.
+        // See https://github.com/flutter/flutter/issues/154842.
```

Comment:
I rewrote this part so that the motivation is more directly explained instead of sounding just like a patch to an issue. How do you think?
```
// Synthesizes remove events immediately after the UP event so that the touches 
// are divided into distinct segments, each beginning with a DOWN event and ending with an UP event. 
// This approach makes sense since each segment can be considered a separate pointer, 
// and it prevents Flutter from generating hover events between these segments, which are 
// not applicable to touch screens, as hovering is not possible.
// (Flutter will automatically synthesize an add event before the pointer makes its next contact.)
```

Reply:
I think this is a better explanation - changed